ush/gfs/task/analysis.py

```diff
@@ -0,0 +1,409 @@
+from gfs.task.base import Task
+import datetime as dt
+import logging
+import os
+import shutil
+
+crtmver = '2.3.0'
+
+class Analysis(Task):
+    """
+    Parent class for global analysis tasks
+    """
+    def __init__(self, config):
+        super().__init__(config)
+        # for now config is assumed to be os.environ
+        self.fv3jedi_fix = config['FV3JEDI_FIX']
+        self.obs_list_yaml = config['OBS_LIST']
+        self.obs_yaml_dir = config['OBS_YAML_DIR']
+        self.berror_dir = config['BERROR_DATA_DIR']
+        self.berror_yaml = config['BERROR_YAML']
+        self.berror_yaml_dir = config['BERROR_YAML_DIR']
+        self.component = config['COMPONENT']
+        self.cdump = config['CDUMP']
+
+    def initialize(self):
+        super().initialize()
+        obs_dict = self.get_obs_dict()
+        self.stage_obs(obs_dict)
+
+    def execute(self):
+        """
+        Note this is generally unused and a shell script
+        """
+        super().execute()
+
+    def finalize(self):
+        super().finalize()
+
+    def get_obs_dict(self):
+        """
+        Get a dictionary of observation files to copy/use
+        based on the specified configuration
+        """
+        import ufsda # temporary until this is in workflow
+        import yaml
+        with open(self.obs_list_yaml, 'r') as yamlopen:
+            obs_list_dict = yaml.safe_load(yamlopen)
+        # need to replace vars
+        obs_list_dict['OBS_YAML_DIR'] = self.obs_yaml_dir
+        obs_list_dict = ufsda.yamltools.replace_vars(obs_list_dict)
+        del obs_list_dict['OBS_YAML_DIR']
+        # need a few extra variables defined
+        obs_list_dict['OBS_DATE'] = self.cdate
+        obs_list_dict['OBS_DIR'] = os.path.join(self.datadir, 'obs')
+        obs_list_dict['OBS_PREFIX'] = os.environ['OPREFIX']
+        obs_list_dict = ufsda.yamltools.parse_config(obs_list_dict)
+        # get observers from master dictionary
+        observers = obs_list_dict['observers']
+        obs_dict = {}
+        for ob in observers:
+            obfile = ob['obs space']['obsdatain']['engine']['obsfile']
+            basename = os.path.basename(obfile)
+            obs_dict[os.path.join(os.environ['COMIN_OBS'], basename)] = obfile
```

Comment:
Questions, not a comment.   Does this line link or copy from `COMIN_OBS` to `basename`?  For development we usually link.  For operations we need to copy.  Does this script support both?  If "yes", how do we toggle between the two?  If "not", can this flexibility be added?